drivers/at: make OK and ERROR replies configurable

aabadie opened the pull request:
<!--
The RIOT community cares a lot about code quality.
Therefore, before describing what your contribution is about, we would like
you to make sure that your modifications are compliant with the RIOT
coding conventions, see https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions.
-->

### Contribution description

This PR provides 2 new macro to the AT command parser. These macros allow to change the expected OK and ERROR replies of an AT module.

I have one that doesn't answer OK or ERROR as the driver expects by default but rather +OK and +ERR.

<!--
Put here the description of your contribution:
- describe which part(s) of RIOT is (are) involved
- if it's a bug fix, describe the bug that it solves and how it is solved
- you can also give more information to reviewers about how to test your changes
-->


### Issues/PRs references

#7084 

<!--
Examples: Fixes #1234. See also #5678. Depends on PR #9876.

Please use keywords (e.g., fixes, resolve) with the links to the issues you
resolved, this way they will be automatically closed when your pull request
is merged. See https://help.github.com/articles/closing-issues-using-keywords/.
-->

toonst commented:
ACK

aabadie commented:
@vincent-d, ok to proxy-ACK @toonst'ACK ?

toonst commented on drivers/at/at.c:
```diff
@@ -141,11 +141,15 @@ ssize_t at_send_cmd_get_lines(at_dev_t *dev, const char *command,
             bytes_left -= res;
-            if ((res == (2 + keep_eol)) && (strncmp(pos, "OK", 2) == 0)) {
+            uint8_t len_ok = strlen(AT_RECV_OK);
+            uint8_t len_error = strlen(AT_RECV_ERROR);
```
I think sizet_t iso uint8_t makes more sense

vincent-d replied:
I'm also wondering if `sizeof() - 1` wouldn't be better (compile time).

And below in the `strncmp()`, shouldn't you check if `len_ok` / `len_error` is `!= 0` ?

aabadie commented:
@vincent-d @toonst, I think your comments are now addressed here.

vincent-d commented:
Murdock complains about signed/unsigned comparisons

vincent-d commented:
casting `res` to  `size_t` seems the best to me, btw.

aabadie commented:
Thanks @vincent-d, I just squashed the commits.

aabadie commented:
@vincent-d, I just realized that I dropped your changes by accident and Murdock is failing again because of the sign compare. I fixed the build issue, not sure if it was your initial proposition.

vincent-d commented:
Re-ACK, please squash

aabadie commented:
squashed, CI is green, merging
